Slim/Menu/BrowseLibrary.pm

```diff
@@ -1555,15 +1556,73 @@ sub _albums {
 			}
 
 			my $extra;
+
+			# Have we got tracks for the year from other albums (ones with a different album year)?
+			my $yearTracks = 0;
+			if (  my $year = (grep(/^year:/, @searchTags))[0] ) {
+				my $onlyAlbumYears = $prefs->get('onlyAlbumYears');
+				if ( !$onlyAlbumYears && $year && !(grep(/^release_type:|^work_id:/, @searchTags)) ) {
+					$year =~ s/^year://;
+					$yearTracks = Slim::Schema::Track::yearTracksNotOnYearAlbums($year, $library_id);
+				}
+				#if not showing track years, ensure "All Songs" doesn't include tracks from albums which don't have the selected year 
+				push @searchTags, "only_album_years:$onlyAlbumYears";
```

Comment:
Do we really have to pass along the value? It's a pref which could be read further down the navigation, too?

Reply:
Yes. I am being stupid!

Reply:
Ok. Yes, passing the value along definitely gives the caller more flexibility.